### PR TITLE
Replace localforage with browser extension storage. Fixes #459

### LINKED ISSUE #459: Account disappeared and can not be created again in Firefox running in private mode
OS: Kubuntu 18.04
Firefox: 73.0.1 (64-Bit)
Floccus: 4.0.2
Counterpart: Nextcloud 18.0.1
Last known working version: ca. one month ago? Not exactly sure...

Firefox is running here exclusively in private mode. The already existing account in floccus disappeared suddenly (noticed today). Unfortunately I can't say when it disappeared, since icon looks like always...
Creating a new account is possible but is not present any more when restarting Firefox.
Also all existing bookmarks seem to have been duplicated in the main bookmark folder. Dunno when this happened because I didn't scroll down the list for a very long time. ;-)

Please drop a note if more information needed.

### PR DESCRIPTION
**Pull request by mlshapiro**
This PR  reverts some changes from da085e0 and 5739a5c.

### CODE REVIEW

**Comment by marcelklehr** [comment]
Thanks for taking this on. Dumping localforage makes me sad in a way, as it is more performant than the normal extension storage, but I guess the main bottleneck is not the storage.

File: src/lib/Controller.js [hunk 1]
```diff
@@ -90,18 +90,6 @@ export default class Controller {
       })
     })
 
-    // migrate from extension storage to indexedDB
-
-    browser.storage.local
-      .get('accounts')
-      .then(async d => {
```
Comment (marcelklehr) [inline src/lib/Controller.js:97]: We'll need another migration like this from localforage to normal storage, so no accounts get lost.

File: src/lib/AccountStorage.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { Folder } from './Tree'
 import AsyncLock from 'async-lock'
 import * as localForage from 'localforage'
 
+const storage = (window.browser && browser.storage) || (window.chrome && chrome.storage);
```
Comment (marcelklehr) [inline src/lib/AccountStorage.js:8]: Please use the browser module https://github.com/marcelklehr/floccus/blob/develop/src/lib/browser-api.js ;)

File: src/lib/Controller.js (hunk 1 above)
Comment (mlshapiro) [reply src/lib/Controller.js:97]: done in 0e9fb1e21b3e6d4e6f3442930f5e2cb84212a7d2

File: src/lib/AccountStorage.js (hunk 2 above)
Comment (mlshapiro) [reply src/lib/AccountStorage.js]: an indeed, thanks for the reference

**Comment by mlshapiro** [comment]
@marcelklehr your requested changes should be handled. Thanks for the review.

File: src/lib/Controller.js [hunk 3]
```diff
@@ -90,16 +91,15 @@ export default class Controller {
...
 
-    // migrate from extension storage to indexedDB
+    // migrate from localForage back to extension storage
 
-    browser.storage.local
-      .get('accounts')
+    localForage.getItem('accounts')
       .then(async d => {
         if (!d.accounts) return
         return AccountStorage.changeEntry('accounts', () => d.accounts)
```
Comment (marcelklehr) [inline src/lib/Controller.js:99]: Gnaaa, this doesn't work. I should have noticed.

Comment (mlshapiro) [reply src/lib/Controller.js:99]: hm, I'm sorry to hear that. I tested the upgrade process using the version of the `develop` branch and had success getting my accounts to update. 

I guess its a problem if you are coming from an older version that had the browser storage to begin with.

**Pull request merged**